### ISSUE TITLE
Deeper view of generic parameters?

### ISSUE THREAD

**Issue by newobj** [ccontrols/structured-types#4]
Code under analysis:

```
export class Generic<T> {
  value: T
  constructor(value: T) {
    this.value = value
  }
}

export class Test {
  make(gen: Generic<boolean>) { return gen.value }
}
```

Analysis yields:

```
{
  Generic: {
    name: 'Generic',
    kind: 13,
    generics: [ { name: 'T' } ],
    properties: [
      { name: 'value', kind: 15, type: 'T' },
      {
        kind: 21,
        parameters: [ { name: 'value', kind: 15, type: 'T' } ]
      }
    ]
  },
  Test: {
    name: 'Test',
    kind: 13,
    properties: [
      {
        name: 'make',
        kind: 11,
        parameters: [
          {
            name: 'gen',
            kind: 13,
            generics: [ { name: 'T' } ],
            properties: [
              { parent: 'Generic', name: 'value', kind: 15, type: 'T' }
            ]
          }
        ],
        returns: { kind: 3 }
      }
    ]
  }
}
```

This tells me that the first parameter ('gen') to Test.make() is the class Generic, and that it has one generic parameter named T. But it doesn't help me see that specifically the type of the 'gen' parameter was Generic&lt;boolean&gt;, e.g. that T was boolean.

Any idea if this is achievable?

**Comment by atanasster** [comment]
No idea for now, will try it tonight. Thanks a lot for the feedback.

**Comment by atanasster** [comment]
I think I got it working, just published in `v3.20.3`

Here is the code in the playground: https://bit.ly/3CDlDO7

Thanks for submitting, I will close it for now pls open new issues if anything is still off
